#!/usr/bin/env python
import argparse
import sys
import os.path
import subprocess

parser = argparse.ArgumentParser(description='Check output file against expected result.')
parser.add_argument('page_number', type=int,
                   help='the number of pages')
parser.add_argument('page_size', type=int,
                   help='the number of unsigned integers in each page')
parser.add_argument('seed', type=int, default=1337,
				   help='random number generator seed')
parser.add_argument('output_value', type=str, 
				   help='output file')

args = parser.parse_args()
__dir__ = os.path.dirname(os.path.realpath(__file__))

try:
	subprocess.check_call(
		[os.path.join(__dir__, "../utilities/data"), 
			str(args.page_number), 
			str(args.page_size),
			str(args.seed),
			"--checkoutputvalue",
			args.file
		], stdout=sys.stdout, stderr=sys.stderr)
	sys.exit(0)
except subprocess.CalledProcessError as e:
	sys.exit(e.returncode)


